### PR TITLE
[74] Implement get all actors that applied for the specific job with jobid API

### PR DESCRIPTION
**Pull request by PKhing**
## What did you do

- Implement get all actors that applied for the specific job with jobid API with unit test
- will be edited later if frontend requires more field

## Demo

- `GET /jobs/{id}/actors` (I recommend you to try this on local, since we don't have any way to apply for job on dev yet)

### CODE REVIEW

File: apps/api/src/modules/job/application/application.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    expect(service).toBeDefined()
+  })
+
+  describe('findByJobId', () => {
+    const MOCK_CASTING_ID = 1
+    const MOCK_JOB_ID = 1
+    const MOCK_ACTORS = mock('user')
+      .pick(['firstName', 'middleName', 'lastName', 'profileImageUrl'])
+      .get(3)
+      .map((actor) => ({ ...actor, actorId: 1, resumeId: 1 }))
```
Comment (palmcm) [inline apps/api/src/modules/job/application/application.service.spec.ts:40]: should use index

**Review by palmcm — CHANGES_REQUESTED** [review]

File: apps/api/src/modules/job/application/application.controller.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  constructor(private readonly applicationService: ApplicationService) {}
+
+  @Get(':id/actors')
+  @UseTypeAuthGuard(UserType.CASTING)
+  @ApiOperation({ summary: 'get all actors that applied in jobs with jobid' })
+  @ApiOkResponse({ type: GetAppliedActorDto })
+  @ApiUnauthorizedResponse({ description: 'User is not login' })
+  @ApiNotFoundResponse({ description: 'Job not found' })
+  @ApiForbiddenResponse({
+    description: 'User is not Casting or User is not the owner of this job',
```
[the remark this answers is not in the document]
Comment (palmcm) [reply apps/api/src/modules/job/application/application.controller.ts:29]: nvm if user is not casting user will not be owner anyway so "or" or "and" it doesn't matter

[the remark this answers is not in the document]
Comment (PKhing) [reply apps/api/src/modules/job/application/application.controller.ts:29]: I assume that UseTypeAuthGuard will throw 403 User is not Casting (should I remove type guard ?)

File: apps/api/src/modules/job/job.module.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import { PrismaModule } from 'src/database/prisma.module'
 
+import { ApplicationModule } from './application/application.module'
 import { JobController } from './job.controller'
 import { JobRepository } from './job.repository'
 import { JobService } from './job.service'
 
 @Module({
-  imports: [PrismaModule],
+  imports: [PrismaModule, ApplicationModule],
```
[the remark this answers is not in the document]
Comment (PKhing) [reply apps/api/src/modules/job/job.module.ts:10]: If I don't import nest won't see ApplicationModule (or should I import at appModule ?)

[the remark this answers is not in the document]
Comment (palmcm) [reply apps/api/src/modules/job/job.module.ts:10]: You should import it in AppModule

[the remark this answers is not in the document]
Comment (palmcm) [reply apps/api/src/modules/job/job.module.ts:10]: But If you say that ApplicationModule should be only use my JobModule you can do this.

**Review by palmcm — APPROVED** [review]
LGTM

**Pull request merged**